### PR TITLE
read collection size from any BsonNumber

### LINKED ISSUE #1374: Bug: could not read collection statistics: Value expected to be of type INT32 is of unexpected type DOUBLE
We're getting this error when the collection size statistic is a Double instead of an Integer when running any query against a large collection.

### PR DESCRIPTION
**Pull request by mossprescott**
Fixes #1374.

### CODE REVIEW

**Comment by mossprescott** [comment]
FYI: http://api.mongodb.com/java/current/org/bson/BsonDocument.html

File: mongodb/src/main/scala/quasar/physical/mongodb/MongoDbIO.scala [hunk 1]
```diff
@@ -288,17 +288,17 @@ object MongoDbIO {
   def collectionStatistics(coll: Collection): MongoDbIO[CollectionStatistics] = {
     val cmd = Bson.Doc(ListMap("collStats" -> coll.collection.bson))
 
-    def intValue(doc: BsonDocument, field: String): String \/ Long =
-      \/.fromTryCatchNonFatal(Option(doc.getInt32(field)).map(_.longValue) \/>
+    def longValue(doc: BsonDocument, field: String): String \/ Long =
+      \/.fromTryCatchNonFatal(Option(doc.getNumber(field)).map(_.longValue) \/>
```
Comment (sellout) [inline mongodb/src/main/scala/quasar/physical/mongodb/MongoDbIO.scala]: So, it’s still an integer, just stored in a Double? Or what is happening here?!

**Comment by mossprescott** [comment]
Note: I have not actually tested this with a collection containing more than 2 GB of data, and don't intend to. Just going on the javadoc, along with the reported error.

**Comment by sellout** [comment]
This raises a question – do we have any way to generate an installer (or even a jar) that we can give to a customer to test this before we merge a publish a new version?

**Comment by mossprescott** [comment]
Not a legit way that I know of. We could certainly build a JAR manually if @boie0025 is willing to test it.

**Comment by sellout** [comment]
Yeah, I think we should at least document somewhere how to build a JAR from a PR and where to tell a customer to copy it, then we can push that closer to test installers.

**Comment by sellout** [comment]
I’ll merge once the build succeeds.

**Comment by jdguzman** [comment]
@sellout @mossprescott we are willing to test if you can provide a repl jar at least. Or I can see if I can build the pr branch.

**Comment by jdguzman** [comment]
FYI I work with Nic

**Comment by boie0025** [comment]
FTR, the collection "size" attribute was 1461660637 at failure.

**Comment by mossprescott** [comment]
@sellout: I think you should go ahead and merge. I think the risk of breakage is low and the easiest way to get jars to test is just to merge it. Make sense?

**Comment by jdguzman** [comment]
@mossprescott @sellout, @boie0025 and myself are going to try building this PR branch and I'll report back what we find.

**Comment by mossprescott** [comment]
@jdguzman: that's great! Let us know if you have any trouble. And welcome to the team ;)

**Comment by jdguzman** [comment]
@mossprescott this is the one that can be confirmed fixed.  We are looking at your other PR now.

**Comment by mossprescott** [comment]
@jdguzman: ok, that makes more sense and thanks again for helping confirm these fixes.

**Comment by jdguzman** [comment]
@mossprescott ok for our particular use case this had a small affect but I'm thinking that the slowness may just be a byproduct of the dataset size and the query.

**Comment by sellout** [comment]
@jdguzman Commenting on the wrong PR again? 😉

**Comment by jdguzman** [comment]
trying to do to many things at once ...

**Comment by sellout** [comment]
Just wanted to make sure before I merged it 😃

**Pull request merged**